WeatherMonitoringSystem/Domain/Bots/BotFactory.cs

```diff
@@ -0,0 +1,25 @@
+﻿
+using WeatherMonitoringSystem.Core;
+using static System.Collections.Specialized.BitVector32;
+
+namespace WeatherMonitoringSystem.Bots;
+
+public static class BotFactory
+{
+    public static List<IWeatherBot> CreateBots(WeatherStation station)
```

Comment:
This method name indicates that it creates bots, but in the method we adding those the station. Maybe would be better to have just an extension class for `WeatherStation` and have a method called `AttachBots` or `AddBots`. Since this is more a logic related to the station.